File: Java_Generated/Java/untitled/src/category_theory/CategoryObject.java
package category_theory;

import category_theory.interfaces.ICategoryObject;
import diagram.interfaces.IDesktop;
import error_handler.interfaces.ICheck;
import error_handler.interfaces.IErrorHandler;
import general_service.Performer;

/**
 *
 */
public class CategoryObject implements ICategoryObject {

    protected  String name;

    public IDesktop desktop;



    public CategoryObject(String name, IDesktop desktop)
    {
        this.name = name;
        this.desktop = desktop;
        if (desktop != null)
        {
            desktop.addCategoryObject(this);
            checker = desktop.getCheck();
            errorHandler = desktop.getErrorHandler();
        }

    }

    /**
     * @return
     */
    @Override
    public String getCategoryObjectName() {
        return name;
    }

    /**
     * @return
     */
    @Override
    public IDesktop getDesktop() {
        return desktop;
    }

    protected boolean check(Object o)
    {
        return checker.check(o);
    }
    protected void show(String message)
    {
        errorHandler.show(message);
    }

    protected void handle(Throwable throwable)
    {
        errorHandler.handle(throwable);
    }

    IErrorHandler errorHandler;

    protected ICheck checker;

    protected Performer performer = new Performer();


}

File: src/Python/category_object.py
from typing import Any
from desktop import Desktop

class CategoryObject:
    def __init__(self, name: str, desktop: Desktop):
        self._name = name
        self._desktop = desktop
        if self._desktop is not None:
            self._desktop += self
            # TODO add checker and error handler
        raise NotImplemented()

    @property
    def name(self):
        return self._name
    
    @property
    def desktop(self):
        return self._desktop
    
    def _check(self, obj: Any) -> bool:
        raise NotImplemented()
    
    def _show(self, message: str) -> None:
        raise NotImplemented()
    
    def _handle(self, exception: Exception) -> None:
        raise NotImplemented


    